strategy/from_json/to_object_containing.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
+
+import { JSONArray, JSONValue, Serializable } from "../../serializable.ts";
+import { FromJSONStrategy } from "../compose_strategy.ts";
+import {
+  ERROR_TO_OBJECT_CONTAINING_INVALID_SUB_VALUE,
+  ERROR_TO_OBJECT_CONTAINING_INVALID_VALUE,
+  ERROR_TO_OBJECT_CONTAINING_USE_TO_SERIALIZABLE,
+} from "../../error_messages.ts";
+
+/** revive data from `{k: v}` using `fromJSON` on a subclass type `v` */
+export function toObjectContaining<T>(
+  type: T & { new (): Serializable },
```

Comment:
I am a bit concerned with the `{ new (): Serializable }` type here as it forces the Object class to not implement a constructor with arguments, which may not be the case for some classes.

![image](https://user-images.githubusercontent.com/49878815/101657211-a01c5b00-3a11-11eb-9980-e457e9a6b920.png)
^ `toObjectContaining` won't accept `SomeClass` as an argument becase it doesn't have a no args constructor

I think an initializer function could work around this:
```ts
toObjectContaining(() => new SomeClass("defaultValue"))
```

Reply:
`toSerializable` would also have the same issue

Reply:
Hmm, well if it's at least consistent we can probably just leave it for now.

We may want to look into an overload or something at some point though so we can support constructors with args. This would help when implementors want to gradually move thier classes over to ts_serialize, rather than having to rewrite all their classes without no-args constructors.

Reply:
fixed it for both `toObjectContaining` an `toSerializable`